var/spack/repos/builtin/packages/py-proj/package.py

```diff
@@ -0,0 +1,28 @@
+from spack import *
+
+class PyProj(Package):
+    """Python interface to the PROJ.4 Library."""
+    homepage = "http://jswhit.github.io/pyproj/"
+    url      = "https://github.com/jswhit/pyproj/tarball/v1.9.5.1rel"
+
+
+    version('1.9.5.1', 'a4b80d7170fc82aee363d7f980279835')
+
+    # We need the benefits of this PR
+    # https://github.com/jswhit/pyproj/pull/54
+    version('citibeth-latlong2',
+        git='https://github.com/citibeth/pyproj.git',
+        branch='latlong2')
+
+    extends('python')
+
+    depends_on('py-cython')
+    depends_on('py-setuptools')
```

Comment:
Please hold this PR, upstream just came through
On May 10, 2016 9:49 AM, "Elizabeth A. Fischer" <
elizabeth.fischer@columbia.edu> wrote:

> @tgamblin: I agree on the code sharing.  But I don't think there's much to
> be done here, short of significant rewrites of `py-proj`.  If there had
> been an obvious way to depend on an external `proj`, I would have done it.
>
> I'm also not comfortable putting a git version in the Spack dev branch,
> because it is not checksummed.  The problem is, the release version of
> py-proj has a small-to-fix, but show-stopping bug (for my application).
> Some PR's were submitted, but remain unmerged.  Unfortunately, the branch
> in my fork is still the only one that works (for me).
>
>     https://github.com/jswhit/pyproj/pull/73
>
> I can see the following ways forward:
>
> 1. Hold this PR until upstream is fixed.  The problem with that is,
> someone else who needs py-proj might waste their effort re-writing this
> Spack package.
>
> 2. Remove the citibeth-latlong2 version from the PR that gets merged into
> dev, and hope that other Spack users don't need it (they probably won't
> unless they use Python3).
>
> 3. Create a tagged release of citibeth-latlong2 on my fork, and add that
> version to Spack, with hashcode.
>
> I'm thinking (3) is probably the way forward.
>
> Your thoughts?
> -- Elizabeth
>
>
> On Tue, May 10, 2016 at 4:39 AM, Todd Gamblin <notifications@github.com>
> wrote:
>
>> In var/spack/repos/builtin/packages/py-proj/package.py:
>>
>> > +    homepage = "http://jswhit.github.io/pyproj/"
>> > +    url      = "https://github.com/jswhit/pyproj/tarball/v1.9.5.1rel"
>> > +
>> > +
>> > +    version('1.9.5.1', 'a4b80d7170fc82aee363d7f980279835')
>> > +
>> > +    # We need the benefits of this PR
>> > +    # https://github.com/jswhit/pyproj/pull/54
>> > +    version('citibeth-latlong2',
>> > +        git='https://github.com/citibeth/pyproj.git',
>> > +        branch='latlong2')
>> > +
>> > +    extends('python')
>> > +
>> > +    depends_on('py-cython')
>> > +    depends_on('py-setuptools')
>>
>> @citibeth <https://github.com/citibeth>: I think it's bad if using
>> py-proj and proj at the same time fails... maybe @mathstuf
>> <https://github.com/mathstuf> can look into this if he's got time.
>> @mathstuf <https://github.com/mathstuf>: have you used an external proj?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/LLNL/spack/pull/717/files/552a2815cd82b04c74eac44352f5e812ea0fa7f6#r62631624>
>>
>
>